log/logger_setup.go

```diff
@@ -1,35 +1,51 @@
 package log
 
 import (
+	"errors"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"os"
 	"strings"
 
 	"github.com/thrasher-corp/gocryptotrader/common/convert"
 )
 
-func getWriters(s *SubLoggerConfig) io.Writer {
-	mw := MultiWriter()
-	m := mw.(*multiWriter) // nolint // type assert not required
+var (
+	errSubloggerConfigIsNil  = errors.New("sublogger config is nil")
+	errUnhandledOutputWriter = errors.New("unhandled output writer")
+)
 
+func getWriters(s *SubLoggerConfig) (io.Writer, error) {
+	if s == nil {
+		return nil, errSubloggerConfigIsNil
+	}
+	mw, err := MultiWriter()
+	if err != nil {
+		return nil, err
+	}
 	outputWriters := strings.Split(s.Output, "|")
 	for x := range outputWriters {
-		switch outputWriters[x] {
+		var writer io.Writer
+		switch strings.ToLower(outputWriters[x]) {
 		case "stdout", "console":
-			m.Add(os.Stdout)
+			writer = os.Stdout
 		case "stderr":
-			m.Add(os.Stderr)
+			writer = os.Stderr
 		case "file":
 			if FileLoggingConfiguredCorrectly {
-				m.Add(GlobalLogFile)
+				writer = GlobalLogFile
 			}
 		default:
-			m.Add(ioutil.Discard)
+			// Note: Do not want to add a ioutil.discard here as this adds
+			// additional routines for every write for no reason.
+			return nil, fmt.Errorf("%w: %s", errUnhandledOutputWriter, outputWriters[x])
```

Comment:
I'd be fine with this if you actually surfaced this error, but if you want to continue to hide/discard the error I'd prefer you to continue to use `Discard`. Otherwise a user has no idea why their entire logger isn't working (see below).

You may also wish to also prevent the application from running on the event of a bad logger

```
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
2021/11/29 09:32:46 Logger write error: io.Writer not set
```